examples/nested/run-tests.sh

```diff
@@ -33,10 +33,6 @@ helm upgrade --install --create-namespace spire charts/spire \
   --values "${DEPS}/spire-root-server-values.yaml" \
   --wait
 
-kubectl get nodes -o go-template='{{range .items}}{{printf "%s\n" .metadata.uid}}{{end}}' | while read -r line; do
-  kubectl exec -t spire-server-0 -n "spire-root-server" -- spire-server entry create -spiffeID spiffe://production.other/production/nested-spire -parentID "spiffe://production.other/spire/agent/k8s_psat/production/$line" -selector k8s:pod-label:app.kubernetes.io/name:server -downstream
-done
-
```

Comment:
Why is this being removed?

Reply:
With the ability to have multiple controller managers, the spire-root server now has its own, and can be configured via 
the spire charts built in ClusterSPIFFEID rather then having to manually load spire entries into the database. Its now configured here:
https://github.com/spiffe/helm-charts-hardened/blob/controller-manager-class/.github/tests/dependencies/spire-root-server-values.yaml#L8-L16

This is much better then the for loop, as the loop is static. It only loads the known k8s nodes available at the time it runs, where as the controller manager will pick up new nodes and remove old nodes as they come and go from the k8s cluster.

Reply:
the downstream setting there is confusing to me? what has a root server to do with a downstream? Shouldn't it be the case nested servers should only refer to upstream servers. To not end up in chicken and egg scenarios?

E.g.

I start with a single spire instance, now later on I want to add a nested one, why would I have to change the upstream deployment in order to get the new nested (downstream) working?

Reply:
Say you have a nested setup. you have essentially:
root spire-server -> root clusterspiffeid config -> upstream spire-agent -> nested spire-server -> nested spire clusterspiffeid config -> downstream spire-agent -> user workload.

nested spire-server is a downstream server of root spire-server. Downstream=true is set on root spire clusterspiffeid so that root spire-server will hand out an intermediate ca identity to the nested spire-server via upstream spire-agent.

nested clusterspiffeid does not have downstream true, so user workload won't get an intermediate ca.

Reply:
Ok it starts to become more clear for me. Guess this is good enough for now. In case we face issues or more questions from community we can always revise to make it more clear. Maybe it is just me 😊